docs/solana/provider-api/methods.md

```diff
@@ -0,0 +1,327 @@
+It takes a
+[solanaWeb3.Transaction](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html)
+and return a promise containing a
+[solanaWeb3.Transaction](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html)
+
+```ts
+solana.signTransaction(solanaWeb3.Transaction) : Promise<solanaWeb3.Transaction>
+```
+Example:
+```ts
+const NETWORK = clusterApiUrl("testnet");
+const connection = new Connection(NETWORK);
+const transaction = new Transaction();
+try {
+  const signedTransaction = await window.solana.signTransaction(transaction);
+  const signature = await connection.sendRawTransaction(signedTransaction.serialize());
+} catch (err) {
+  // { code: 4001, message: 'User rejected the request.' }
+}
+```
+
+:::danger Deprecated API
+
+This API is deprecated but is available in Brave Wallet.
+
+:::
+
+## `solana.signAllTransactions` (deprecated)
+
+This API allows a website to sign some transactions and submit them later.
+It takes an array of
+[solanaWeb3.Transaction](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html)
+and return a promise containing an array of
+[solanaWeb3.Transaction](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html)
+
+```ts
+solana.signTransaction(solanaWeb3.Transaction[]) : Promise<solanaWeb3.Transaction[]>
+```
+Example:
+```ts
+const NETWORK = clusterApiUrl("testnet");
+const connection = new Connection(NETWORK);
+const transaction1 = new Transaction();
+const transaction2 = new Transaction();
+try {
+  const signedTransactions = await window.solana.signTransaction([transaction1, transaction2]);
+  signedTransactions.forEach(signedTransaction => {
+    const signature = await connection.sendRawTransaction(signedTransaction.serialize())});
+} catch (err) {
+  // { code: 4001, message: 'User rejected the request.' }
+}
+```
+:::danger Deprecated API
+
+This API is deprecated but is available in Brave Wallet.
+
+:::
+
+## `solana.signMessage`
+
+Allows a website to sign a message using selected account. If we detect the
+message payload to be signed is a transaction, we will reject the request
+automatically.
+This API takes **Uint8Array** with optional encoding (**hex** or **utf8**) to
+display the message to users. If a caller doesn't specify encoding, default
+would be utf8.
+
+```ts
+solana.signMessage(Uint8Array, string?) : Promise({ publicKey: <solanaWeb3.PublicKey>,
+                                                    signature: <Uint8Array> })
+ ```
+
+ Example:
+ ```ts
+ const encodedMessage = new TextEncoder().encode('signMessage test');
+ // signedMessage1 === signedMessage2 === signedMessage3
+ const signedMessage1 = await window.solana.signMessage(encodedMessage);
+ const signedMessage2 = await window.solana.signMessage(encodedMessage, "utf8");
+ const signedMessage2 = await window.solana.signMessage(encodedMessage, "hex");
+ ```
+
+## `solana.request`
+
+Request API allow a website call above methods with an universal interface.
```

Comment:
The `request` API allows a website to call the above methods with a universal interface.